packages/features/bookings/lib/handleNewBooking/ensureAvailableUsers.ts

```diff
@@ -64,11 +69,46 @@ export async function ensureAvailableUsers(
 
   const bookingLimits = parseBookingLimit(eventType?.bookingLimits);
   const durationLimits = parseDurationLimit(eventType?.durationLimits);
-
+  let attendeeUsers: IsFixedAwareUser[] = [];
+  let isEventOwnerOrTeamAdmin = false;
+  if (input.originalRescheduledBooking?.uid) {
+    const isEventOwner = isUserReschedulingOwner(userId, input.originalRescheduledBooking?.user?.id);
+    const membership = eventType?.team?.members.find((membership) => membership.userId === userId);
+    const isUserTeamAdminOrOwner =
+      membership?.role === MembershipRole.OWNER || membership?.role === MembershipRole.ADMIN;
+    isEventOwnerOrTeamAdmin = isEventOwner || isUserTeamAdminOrOwner
+    if (isEventOwnerOrTeamAdmin) {
+      const attendeesEmails = input.originalRescheduledBooking?.attendees?.map((a) => a.email);
+      if (attendeesEmails?.length) {
+        let attendees = await prisma.user.findMany({
+          where: {
+            email: {
+              in: attendeesEmails,
+            },
+          },
+          select: {
+            ...userSelect.select,
+            credentials: {
+              select: credentialForCalendarServiceSelect,
+            },
+          }
+        });
+        attendeeUsers = attendees.map((u) => {
+          return {
+            ...u,
+            isFixed: true,
+          };
+        });
+      }
+    }
+  }
+  const totalUsers = [...eventType.users, ...attendeeUsers].filter(
+    (user, index, self) => index === self.findIndex((t) => t.id === user.id)
+  );
```

Comment:
`EnsureAvailableUsers` takes users and checks if those users are present for the timeslot. It is not good practice to modify those users inside the function. This will make it prone to errors in future.

Reply:
Made the change